src/python/pants/goal/run_tracker.py

```diff
@@ -503,3 +518,18 @@ def _get_option_to_record(self, scope, option=None):
             raise ValueError(
                 f"Couldn't find option scope {scope}{option_str} for recording ({e!r})"
             )
+
+    def retrieve_logs(self) -> List[str]:
```

Comment:
This should probably be `Tuple[str, ...]`.

Reply:
meh.. copying memory... this is not free.